### PR TITLE
Refactor NUnit v3 support

### PR DESCRIPTION
**Pull request by ngyukman**
- Changed the nunit version specific implementation into mixins
- Added getCommandArgs to support gradle-opencover-plugin
- Renamed parallel_forks to parallelForks to follow the same naming
- Default working directory of NUnit v3 to build\nunit\

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 # gradle-nunit-plugin changelog
 
 ## 1.6
+### Added
+* `getCommandArgs` is added to NUnit task
+
+### Changed
+* Default working directory of NUnit v3 to build\nunit\
+* Renamed parallel_forks to parallelForks in NUnit task
```
Comment (timotei) [inline CHANGELOG.md:9]: :+1: good idea

File: README.md [hunk 2]
```diff
@@ -25,6 +25,7 @@ It creates a task 'nunit' that may be configured as follows:
 
     nunit {
         // optional - defaults to '2.6.4', but plugin is compatible with v3+ as well
+        // it must be the *first* parameter if specified
```
Comment (timotei) [inline README.md]: That's a weird requirement, no?

Comment (ngyukman) [inline README.md]: I agree
Actually it is only required when you specific version specific settings and I think it is valid reason for that
I could update the comment to state it clearer

Comment (florian-gandon) [inline README.md]: +1

File: src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy [hunk 3]
```diff
@@ -15,29 +15,21 @@ class NUnit extends ConventionTask {
...
-    def testList
-    def test
-
     def reportFolder
     boolean useX86 = false
     boolean shadowCopy = false
     String reportFileName = 'TestResult.xml'
     boolean ignoreFailures = false
-    boolean parallel_forks = true
+    boolean parallelForks = true
```
Comment (timotei) [inline src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy:29]: Breaking change? :D (But I agree on it)

File: src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy [hunk 4]
```diff
@@ -15,29 +15,21 @@ class NUnit extends ConventionTask {
...
+
+    def test
```
Comment (timotei) [inline src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy:31]: If we broke some stuff already, how about we use a single variable, which is always a list? That will include the case when there's a single item.

Comment (ngyukman) [inline src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy:31]: actually test and testList are two different type
-test is a comma-separated list of test to run
-testList is a file path which specify the tests to run

https://github.com/nunit/docs/wiki/Console-Command-Line

Comment (timotei) [inline src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy:31]: Thanks for the link!

File: src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy [hunk 5]
```diff
@@ -74,6 +69,7 @@ class NUnit extends ConventionTask {
     @OutputFile
     File getTestReportPath() {
         // for the non-default nunit tasks, ensure we write the report in a separate file
+        // TODO: Do we need to supply prefix when user has specified its own report file name?
```
Comment (timotei) [inline src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy:72]: Yeah, we probably shouldn't do that :)

**Comment by timotei** [comment]
LGTM

**Pull request merged**